### PR TITLE
Demo data for Leave Allocation, Leave Application and Attendance

### LINKED ISSUE #5730: Attendance and Leave
- [x] Leave Allocation
- [x] Leave Application
- [x] Attendance

### PR DESCRIPTION
**Pull request by KanchanChauhan**
Fixed https://github.com/frappe/erpnext/issues/5730

### CODE REVIEW

File: erpnext/demo/user/hr.py [hunk 1]
```diff
@@ -120,4 +121,42 @@ def make_sales_invoice_for_timesheet(name):
...
+	employee = get_random("Employee")
+	leave_type = frappe.get_all("Leave Type", fields=['name', 'max_days_allowed'])[0]
+	if leave_type.max_days_allowed:
+		max_leave = leave_type.max_days_allowed
+	else:
+		max_leave = 10	
+	leave_type = leave_type.name
+	
+	make_leave_allocation(max_leave, employee, leave_type)
+
```
Comment (rmehta) [inline erpnext/demo/user/hr.py]: no need to allocate leaves every time.

Maybe allocate leaves once (in setup?)

File: erpnext/demo/user/hr.py [hunk 2]
```diff
@@ -123,4 +128,48 @@ def make_sales_invoice_for_timesheet(name):
...
+		if leave_balance != 0:
+			if leave_balance == 1:
+				to_date = nowdate()
+			else:
+				to_date = add_days(nowdate(), random.randint(0, leave_balance-1))
+				
+			leave_application = frappe.get_doc({
+				"doctype": "Leave Application",
+				"employee": allocated_leave.employee,
+				"from_date": nowdate(),
```
Comment (nabinhait) [inline erpnext/demo/user/hr.py]: from_date should be `frappe.flags.current_date`

File: erpnext/demo/user/hr.py [hunk 3]
```diff
@@ -123,4 +128,48 @@ def make_sales_invoice_for_timesheet(name):
...
+			except (OverlapError):
+				frappe.db.rollback()
+			
+def mark_attendance():
+	for employee in frappe.get_all('Employee', fields=['name'], filters = {'status': 'Active'}):
+		if not frappe.db.get_value("Attendance", {"employee": employee.name, "att_date": nowdate()}):
+			attendance = frappe.get_doc({
+				"doctype": "Attendance",
+				"employee": employee.name,
+				"att_date": nowdate()
```
Comment (nabinhait) [inline erpnext/demo/user/hr.py]: Use `frappe.flags.current_date`

File: erpnext/demo/user/hr.py [hunk 4]
```diff
@@ -123,4 +128,48 @@ def make_sales_invoice_for_timesheet(name):
...
+			try:
+				leave_application.insert()
+				leave_application.submit()
+				frappe.db.commit()
+			except (OverlapError):
+				frappe.db.rollback()
+			
+def mark_attendance():
+	for employee in frappe.get_all('Employee', fields=['name'], filters = {'status': 'Active'}):
+		if not frappe.db.get_value("Attendance", {"employee": employee.name, "att_date": nowdate()}):
```
Comment (nabinhait) [inline erpnext/demo/user/hr.py]: frappe.flags.current_date

File: erpnext/demo/setup_data.py [hunk 5]
```diff
@@ -318,6 +319,26 @@ def setup_salary_structure_for_timesheet():
...
+		leave_types = frappe.get_all("Leave Type", fields=['name', 'max_days_allowed'])
+		for leave_type in leave_types:
+			if not leave_type.max_days_allowed:
+				leave_type.max_days_allowed = 10
+	
+		leave_allocation = frappe.get_doc({
+			"doctype": "Leave Allocation",
+			"employee": employee.name,
+			"from_date": nowdate(),
+			"to_date": add_years(nowdate(), 1),
```
Comment (nabinhait) [inline erpnext/demo/setup_data.py]: Allocate leave for the year_start_date and year_end_date for the current year

**Pull request closed without merging**